doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,430 @@ processed in this order:
 - If the value is a sequence, its elements are recursively imported in the
   order they appear.
 
+.. _west-manifest-groups:
+
+Project Groups and Active Projects
+**********************************
+
+You can use the ``groups`` keys briefly described :ref:`above
+<west-manifest-files>` to put projects into groups. The ``groups`` key can be a
+project or top level manifest attribute, like this:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: some-project
+         groups: ...
+     groups: ...
+
+You can allow or block project groups. Projects whose groups are all blocked
```

Comment:
I wonder if we should use "enable" and "disable" instead of "allow" and "block". To me the former feels truer to the original intent, but this is just a suggestion.

Reply:
Same for other instances below

Reply:
I'm trying to adopt 'allow' and 'block' as consistent replacements for 'white' and 'black' in general here, so I'd prefer to keep as-is if that's OK.

Reply:
Yes, but isn't enabling or disabling groups conceptually different from allowing or blocking projects? From my perspective it is, because projects are allowed or blocked directly, whereas enabling or disabling groups then has an impact on what projects end up being active or inactive, which is different from them being allowed or blocked.

Reply:
when using `active` and `inactive`, then I feel `enable` and `disable` are fitting better.
But that's just my feeling as non-native English speaker.

Reply:
> Yes, but isn't enabling or disabling groups conceptually different from allowing or blocking projects? 

This question confused me a lot because you crossed the adjectives and the names. Did you do that on purpose to illustrate your own confusion?

Here's my non-native speaker perspective too:
- The adjectives active and inactive sound (to me) like "final states", they do not like some indirect desire that gets combined with and can takes precedence over others desires/groups. So it's perfect for projects.
- The verbs allow/block do sound like indirect desires. So they're good for groups.
- At first I found the verbs enable/disable sounded better than allow/block because they don't have any security connotation, so they reflect the intent better. However they feel less "indirect" to me, maybe because it's more common to combine security permissions or because they're very close to the adjective enabled/disabled which sound like final states.

Reply:
Oops, I just realized "allow/block" is used for groups in group filters but the same terms are use for projects in `import` :-(

> trying to adopt  'allow' and 'block' as consistent replacements for 'white' and 'black' in general here

I think that particular  "migration consistency" should have lower priority than all other "self-consistencies". What is its purpose, make search/replace easier? That would sound like a small and short-term gain.

Reply:
I am not sure myself. I stand by the comments I've made, and @marc-hb and @tejlmand seem to align, but I can't say that I am convinced. Let me try to summarize so I can try to clarify it for myself and others:

1. Keep as-is:
   - groups can be allowed or blocked
   - projects are active or inactive
   - projects can be placed in an allowlist or blocklist when importing
2. Change to:
   - groups can be enabled or disabled
   - projects are active or inactive
   - projects can be placed in an allowlist or blocklist when importing

I think I prefer 2., because allowing or blocking has a ring to "external". You allow or block something that is defined externally and then you import or deal with. So that is very good for importing, because you are either allowing or blocking a project that is not controlled by you. But groups are described and managed in-manifest, so that is why I think enabling and disabling them makes more sense that allowing them or blocking them.

Reply:
> * Keep as-is:
>   
>   * groups can be allowed or blocked
>   * projects are active or inactive
>   * projects can be placed in an allowlist or blocklist when importing
> 
> * Change to:
>   
>   * groups can be enabled or disabled
>   * projects are active or inactive
>   * projects can be placed in an allowlist or blocklist when importing

Thanks for this.
Thinking a bit more on this, I start to like 2. better and better, as it removes any ambiguity regarding `allow` / `block`.
`allow` is solely used in relation to import (`allowlist`), so will never be mixed up with groups in any way, as groups are enabled or disabled.

Reply:
OK, option 2. it is. I've updated to the documentation to match the code changes in https://github.com/zephyrproject-rtos/west/pull/466.